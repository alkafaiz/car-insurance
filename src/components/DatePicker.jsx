import React, { forwardRef } from "react";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import PropTypes from "prop-types";
import {
  FormControl,
  FormLabel,
  Input,
  FormErrorMessage
} from "@chakra-ui/react";

function CustomDatePicker({
  label,
  isRequired,
  placeholder,
  value,
  onChange,
  error,
  isInvalid,
  inputProps,
  minDate,
  maxDate
}) {
  const CustomInput = forwardRef(({ value, onClick, inputProps }, ref) => (
    <Input
      {...inputProps}
      onClick={onClick}
      ref={ref}
      value={value}
      placeholder={placeholder}
      readOnly
    />
  ));

  return (
    <FormControl isInvalid={isInvalid} isRequired={isRequired} mt={4}>
      <FormLabel>{label}</FormLabel>
      <DatePicker
        wrapperClassName="datepicker-container"
        selected={value}
        onChange={onChange}
        customInput={<CustomInput inputProps={inputProps} />}
        minDate={minDate}
        maxDate={maxDate}
      />
      <FormErrorMessage>{error}</FormErrorMessage>
    </FormControl>
  );
}

CustomDatePicker.defaultProps = {
  isRequired: false,
  placeholder: ""
};

CustomDatePicker.propTypes = {
  onChange: PropTypes.func,
  isInvalid: PropTypes.bool,
  error: PropTypes.string,
  inputProps: PropTypes.object,
  label: PropTypes.string.isRequired,
  isRequired: PropTypes.bool,
  placeholder: PropTypes.string,
  minDate: PropTypes.object,
  maxDate: PropTypes.object
};

export default CustomDatePicker;
