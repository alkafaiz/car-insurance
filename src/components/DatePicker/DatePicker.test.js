import { render } from "@testing-library/react";
import DatePicker from "./index";
import React from "react";

const defaultProps = {
  label: "Date picker"
};

const setup = props =>
  render(<DatePicker {...{ ...defaultProps, ...props }} />);

describe("DatePicker component", () => {
  it("should render", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("render without crashing", () => {
    const wrapper = setup();
    expect(wrapper).toBeDefined();
    expect(wrapper).toBeTruthy();
  });

  it("should render label", () => {
    const labelText = "Testing label";
    const { getByText } = setup({ label: labelText });

    const label = getByText(labelText);
    expect(label).toBeDefined();
    expect(label.textContent).toBe(labelText);
  });

  it("should render error text", () => {
    const errorText = "Required";
    const { getByTestId } = setup({ isInvalid: true, error: errorText });

    const error = getByTestId("error-text");
    expect(error).toBeDefined();
    expect(error.textContent).toBe(errorText);
  });
});
