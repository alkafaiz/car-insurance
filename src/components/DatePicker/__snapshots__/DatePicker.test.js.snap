// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <div
        class="chakra-form-control css-i55s4y"
        role="group"
      >
        <label
          class="chakra-form__label css-5nk1it"
          for="field-1"
          id="field-1-label"
        >
          Date picker
        </label>
        <div
          class="react-datepicker-wrapper datepicker-container"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              aria-describedby="field-1-feedback"
              class="chakra-input css-1u8qly9"
              id="field-1"
              placeholder=""
              readonly=""
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-form-control css-i55s4y"
      role="group"
    >
      <label
        class="chakra-form__label css-5nk1it"
        for="field-1"
        id="field-1-label"
      >
        Date picker
      </label>
      <div
        class="react-datepicker-wrapper datepicker-container"
      >
        <div
          class="react-datepicker__input-container"
        >
          <input
            aria-describedby="field-1-feedback"
            class="chakra-input css-1u8qly9"
            id="field-1"
            placeholder=""
            readonly=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
